engine/storage/datamotion/src/main/java/org/apache/cloudstack/storage/motion/StorageSystemDataMotionStrategy.java

```diff
@@ -1991,8 +2198,49 @@ private void verifyLiveMigrationMapForKVM(Map<VolumeInfo, DataStore> volumeDataS
                 throw new CloudRuntimeException("Destination storage pool with ID " + dataStore.getId() + " was not located.");
             }
 
+            if (storageTypeConsistency == null) {
```

Comment:
Won't this expression always evaluate to true?

Reply:
Actually just for the first item in the map entry set, as the variable is defined out of the `for` block. The idea is to set it on the first iteration and compare it on the next ones.

Reply:
Ah, yes, my mistake. The code view had a section collapsed and I didn't see the looping.

OK, I've gone through the contents of the PR and it LGTM. Thanks!